src/factory.rs

```diff
@@ -1,26 +1,108 @@
-use crate::runtime;
+use crate::runtime::*;
 use crate::*;
 
-// TODO: this should return `Result<&I>` e.g. a reference pointing to the factory cache.
-// So this function needs to be implemented as some sort of atomic/singleton where RoGetActivationFactory
-// is only called once and the result is then cached. Here's how I do it in C++ - it's critical
-// that this is super fast. Also, load RoGetActivationFactory dynamically and fall back to LoadLibrary
-// and implement DLL garbage collection for those. Version 0.1 can probably just pin everything.
-// https://github.com/microsoft/cppwinrt/blob/master/strings/base_activation.h
+/// Attempts to load the factory interface for the given WinRT class.
+///
+/// Note that factory caching still needs to be implemented.
 pub fn factory<C: RuntimeName, I: ComInterface>() -> Result<I> {
-    let mut ptr = std::ptr::null_mut();
+    let mut factory = std::ptr::null_mut();
+    let name = HString::from(C::NAME);
+
     unsafe {
```

Comment:
I have a single unsafe block because this function is riddled with "unsafe" calls. It didn't seem to add value to add the unsafe block around each call.

Reply:
That's fine. At some point in the future it would be nice to have safe  versions of the win32 APIs...

Reply:
Definitely.